### PR TITLE
Settings menu

### PR DESCRIPTION
**Pull request by smaug14**

### CODE REVIEW

**Comment by Adesii** [comment]
Looks good. 
tho we should probably have a predefined list of resolutions defined in the script and then only populate the dropdown with resolutions that actually are supported by getting the screen size from the Displayserver 
[using this](https://docs.godotengine.org/en/stable/classes/class_displayserver.html#class-displayserver-method-screen-get-size)

File: Scripts/UI/Settings/Settings.gd [hunk 1]
```diff
@@ -0,0 +1,35 @@
+extends Control
+
+@onready var resoultion_button: OptionButton = $SettingsContainer/ResolutionContainer/ResolutionSelection
```
Comment (Adesii) [inline Scripts/UI/Settings/Settings.gd:3]: spelling mistake

**Review by Adesii — CHANGES_REQUESTED** [review]

File: Scripts/UI/Settings/Settings.gd (hunk 1 above)
Comment (smaug14) [reply Scripts/UI/Settings/Settings.gd]: Fixed

**Comment by smaug14** [comment]
> Looks good. tho we should probably have a predefined list of resolutions defined in the script and then only populate the dropdown with resolutions that actually are supported by getting the screen size from the Displayserver [using this](https://docs.godotengine.org/en/stable/classes/class_displayserver.html#class-displayserver-method-screen-get-size)

Done, should be only using sensible resolutions for windowed mode

**Comment by smaug14** [comment]
Also fixed warnings that were shown in debugger

**Review by Adesii — APPROVED** [review]

**Pull request merged**